{
  "uid" : "5f38d3b80907c125",
  "name" : "少参code：新增28s测试研究部3分队失败",
  "fullName" : "testcase.test_department_manage.Test_department_manage#test_add_department",
  "historyId" : "59b456ec847beec9ef31e55899b3fe6d",
  "time" : {
    "start" : 1632144077025,
    "stop" : 1632144077098,
    "duration" : 73
  },
  "status" : "failed",
  "statusMessage" : "AssertionError",
  "statusTrace" : "self = <testcase.test_department_manage.Test_department_manage object at 0x00000160EEABB1C0>\nenvironment_url = 'http://ihrm-test.itheima.net'\nget_token = {'Authorization': 'Bearer 6935b6d2-6e87-4949-aeda-272af78e6460'}\ntitle = '少参code：新增28s测试研究部3分队失败'\nask = {'introduce': '一个很垃圾的研究部', 'name': '28s测试研究部3分队', 'pid': 9526}\nexcepted = {'code': 99999, 'message': '请稍后重试', 'status_code': 200, 'success': False}\n\n    @allure.story('新增部门信息-数据驱动测试')\n    @allure.title('{title}')\n    @pytest.mark.usefixtures('re_login')\n    @pytest.mark.parametrize('title, ask, excepted', My_return_data.return_test_data('add_department.yaml'))\n    def test_add_department(self, environment_url, get_token, title, ask, excepted):\n        try:\n            res = Department_Manage().add_department(environment_url, get_token, ask)\n            common_assert(res, excepted)\n        except Exception as e:\n            err = str(res.json()) + '\\n' + str(e)\n            MyLog.error(err)\n>           raise e\n\ntest_department_manage.py:73: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\ntest_department_manage.py:69: in test_add_department\n    common_assert(res, excepted)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nres = <Response [200]>\nexpected = {'code': 99999, 'message': '请稍后重试', 'status_code': 200, 'success': False}\n\n    @allure.step('进行断言')\n    def common_assert(res, expected):\n        '''\n        对响应体进行断言\n        :param res: 请求响应体\n        :param expected: 字典：{success: true， code: 10000， message: '操作成功'， status_code: 200}\n        :return:assert断言\n        '''\n        assert expected.get('status_code') == res.status_code\n>       assert expected.get('code') == res.json().get('code')\nE       AssertionError\n\n..\\utils\\commen_assert.py:13: AssertionError",
  "flaky" : false,
  "newFailed" : false,
  "beforeStages" : [ {
    "name" : "从config.yaml获取环境IP",
    "time" : {
      "start" : 1632144076946,
      "stop" : 1632144076946,
      "duration" : 0
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "hasContent" : false,
    "shouldDisplayMessage" : false,
    "attachmentsCount" : 0
  }, {
    "name" : "获取token",
    "time" : {
      "start" : 1632144077020,
      "stop" : 1632144077024,
      "duration" : 4
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "hasContent" : false,
    "shouldDisplayMessage" : false,
    "attachmentsCount" : 0
  }, {
    "name" : "前置登陆操作",
    "time" : {
      "start" : 1632144076946,
      "stop" : 1632144077020,
      "duration" : 74
    },
    "status" : "passed",
    "steps" : [ {
      "name" : "登陆操作",
      "time" : {
        "start" : 1632144076946,
        "stop" : 1632144077019,
        "duration" : 73
      },
      "status" : "passed",
      "steps" : [ ],
      "attachments" : [ ],
      "parameters" : [ {
        "name" : "url",
        "value" : "'http://ihrm-test.itheima.net'"
      }, {
        "name" : "login_data",
        "value" : "{'mobile': '13800000002', 'password': '123456'}"
      } ],
      "stepsCount" : 0,
      "hasContent" : true,
      "shouldDisplayMessage" : false,
      "attachmentsCount" : 0
    } ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 1,
    "hasContent" : true,
    "shouldDisplayMessage" : false,
    "attachmentsCount" : 0
  } ],
  "testStage" : {
    "status" : "failed",
    "statusMessage" : "AssertionError",
    "statusTrace" : "self = <testcase.test_department_manage.Test_department_manage object at 0x00000160EEABB1C0>\nenvironment_url = 'http://ihrm-test.itheima.net'\nget_token = {'Authorization': 'Bearer 6935b6d2-6e87-4949-aeda-272af78e6460'}\ntitle = '少参code：新增28s测试研究部3分队失败'\nask = {'introduce': '一个很垃圾的研究部', 'name': '28s测试研究部3分队', 'pid': 9526}\nexcepted = {'code': 99999, 'message': '请稍后重试', 'status_code': 200, 'success': False}\n\n    @allure.story('新增部门信息-数据驱动测试')\n    @allure.title('{title}')\n    @pytest.mark.usefixtures('re_login')\n    @pytest.mark.parametrize('title, ask, excepted', My_return_data.return_test_data('add_department.yaml'))\n    def test_add_department(self, environment_url, get_token, title, ask, excepted):\n        try:\n            res = Department_Manage().add_department(environment_url, get_token, ask)\n            common_assert(res, excepted)\n        except Exception as e:\n            err = str(res.json()) + '\\n' + str(e)\n            MyLog.error(err)\n>           raise e\n\ntest_department_manage.py:73: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\ntest_department_manage.py:69: in test_add_department\n    common_assert(res, excepted)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nres = <Response [200]>\nexpected = {'code': 99999, 'message': '请稍后重试', 'status_code': 200, 'success': False}\n\n    @allure.step('进行断言')\n    def common_assert(res, expected):\n        '''\n        对响应体进行断言\n        :param res: 请求响应体\n        :param expected: 字典：{success: true， code: 10000， message: '操作成功'， status_code: 200}\n        :return:assert断言\n        '''\n        assert expected.get('status_code') == res.status_code\n>       assert expected.get('code') == res.json().get('code')\nE       AssertionError\n\n..\\utils\\commen_assert.py:13: AssertionError",
    "steps" : [ {
      "name" : "添加部门",
      "time" : {
        "start" : 1632144077025,
        "stop" : 1632144077097,
        "duration" : 72
      },
      "status" : "passed",
      "steps" : [ ],
      "attachments" : [ ],
      "parameters" : [ {
        "name" : "url",
        "value" : "'http://ihrm-test.itheima.net'"
      }, {
        "name" : "token",
        "value" : "{'Authorization': 'Bearer 6935b6d2-6e87-4949-aeda-272af78e6460'}"
      }, {
        "name" : "add_data",
        "value" : "{'name': '28s测试研究部3分队', 'introduce': '一个很垃圾的研究部', 'pid': 9526}"
      } ],
      "stepsCount" : 0,
      "hasContent" : true,
      "shouldDisplayMessage" : false,
      "attachmentsCount" : 0
    }, {
      "name" : "进行断言",
      "time" : {
        "start" : 1632144077097,
        "stop" : 1632144077097,
        "duration" : 0
      },
      "status" : "failed",
      "statusMessage" : "AssertionError\n",
      "statusTrace" : "  File \"E:\\Python\\Python38\\lib\\site-packages\\allure_commons\\_allure.py\", line 172, in impl\n    return func(*a, **kw)\n  File \"D:\\IHRM_project\\utils\\commen_assert.py\", line 13, in common_assert\n    assert expected.get('code') == res.json().get('code')\n",
      "steps" : [ ],
      "attachments" : [ ],
      "parameters" : [ {
        "name" : "res",
        "value" : "<Response [200]>"
      }, {
        "name" : "expected",
        "value" : "{'success': False, 'code': 99999, 'message': '请稍后重试', 'status_code': 200}"
      } ],
      "stepsCount" : 0,
      "hasContent" : true,
      "shouldDisplayMessage" : true,
      "attachmentsCount" : 0
    } ],
    "attachments" : [ {
      "uid" : "3d4e029a40d7c63a",
      "name" : "log",
      "source" : "3d4e029a40d7c63a.txt",
      "type" : "text/plain",
      "size" : 561
    } ],
    "parameters" : [ ],
    "stepsCount" : 2,
    "hasContent" : true,
    "shouldDisplayMessage" : true,
    "attachmentsCount" : 1
  },
  "afterStages" : [ ],
  "labels" : [ {
    "name" : "epic",
    "value" : "基本信息管理模块"
  }, {
    "name" : "feature",
    "value" : "部门管理模块"
  }, {
    "name" : "story",
    "value" : "新增部门信息-数据驱动测试"
  }, {
    "name" : "tag",
    "value" : "@pytest.mark.usefixtures('re_login')"
  }, {
    "name" : "parentSuite",
    "value" : "testcase"
  }, {
    "name" : "suite",
    "value" : "test_department_manage"
  }, {
    "name" : "subSuite",
    "value" : "Test_department_manage"
  }, {
    "name" : "host",
    "value" : "LAPTOP-LM9EFOVP"
  }, {
    "name" : "thread",
    "value" : "9360-MainThread"
  }, {
    "name" : "framework",
    "value" : "pytest"
  }, {
    "name" : "language",
    "value" : "cpython3"
  }, {
    "name" : "package",
    "value" : "testcase.test_department_manage"
  }, {
    "name" : "resultFormat",
    "value" : "allure2"
  } ],
  "parameters" : [ {
    "name" : "ask",
    "value" : "{'name': '28s测试研究部3分队', 'introduce': '一个很垃圾的研究部', 'pid': 9526}"
  }, {
    "name" : "excepted",
    "value" : "{'success': False, 'code': 99999, 'message': '请稍后重试', 'status_code': 200}"
  }, {
    "name" : "title",
    "value" : "'少参code：新增28s测试研究部3分队失败'"
  } ],
  "links" : [ ],
  "hidden" : true,
  "retry" : true,
  "extra" : {
    "categories" : [ ],
    "tags" : [ "@pytest.mark.usefixtures('re_login')" ]
  },
  "source" : "5f38d3b80907c125.json",
  "parameterValues" : [ "{'name': '28s测试研究部3分队', 'introduce': '一个很垃圾的研究部', 'pid': 9526}", "{'success': False, 'code': 99999, 'message': '请稍后重试', 'status_code': 200}", "'少参code：新增28s测试研究部3分队失败'" ]
}